fe/fe-core/src/main/java/org/apache/doris/statistics/ColumnStat.java

```diff
@@ -54,7 +51,7 @@
  * For example:
  * "@ndv = 10" means that the number distinct values is 10 in the whole table.
  */
-public class ColumnStats {
+public class ColumnStat {
```

Comment:
Would it be better to use `stats`? the meaning of `stat` is not clear, and the abbreviation of statistics is usually `stats`.

Reply:
Yes, I studied some other systems and `stats` (postgres, noisipage, tidb)is do more common used than `stat`(Spark), change it to `stat` is due to the name style of nereids that for the name of variable with collections type should have a `s` as suffix. Andi t looks kind of wired to name a varibale as `columnStatss`.

Reply:
I feel this is not necessary, we'd better try to keep it consistent with other databases.`stat` feels more like the meaning of status, `stats` means statistical information,
And `ColumnStats` is used to describe the statistics of the original table column.

For the problem of naming the collection of nereids, the columnStats of the collection type uses `columnsStats`,
Instead of `columnStatss`, does this also work? Of course, I think these can be discussed further.

I roughly looked at the overall code, and I found that the Stats-Derivation module may only temporarily support `Double` type statistics, so you changed the type of columnStats to
`Double` type.

What I understand is that Statistics is used to describe the statistical information of the original table, and Stats-Derivation is used to derive statistics after each operator. They actually belong to two independent
module. Stats-Derivation obtains the statistics of the original table from Statistics, and should not modify the statistics of the original table. The original statistics can only be modified by specific tasks or statements.

So, should the type support of Stats-Derivation be adapted in the Stats Derivation module? In addition, this means that the previously designed and implemented column statistics need to be reimplemented one by one from the `Double` type. I think this is unreasonable, for example, users can view other types of statistics, and because this change only allows viewing of `Double` type stats.

Reply:
I roughly looked at the overall code, and I found that the Stats-Derivation module may only temporarily support `Double` type statistics, so you changed the type of columnStats to
`Double` type.

What I understand is that Statistics is used to describe the statistical information of the original table, and Stats-Derivation is used to derive statistics after each operator. They actually belong to two independent
module. Stats-Derivation obtains the statistics of the original table from Statistics, and should not modify the statistics of the original table. The original statistics can only be modified by specific tasks or statements.

So, should the type support of Stats-Derivation be adapted in the Stats Derivation module? In addition, this means that the previously designed and implemented column statistics need to be reimplemented one by one from the `Double` type. I think this is unreasonable, for example, users can view other types of statistics, and because this change only allows viewing of `Double` type stats.

Reply:
1. Is there any system support non-numric type to do the stats calculation?  Since cannot find a proper way to map a string value or some other type to a limited numric range
2. I agreed that it shouldn't be modified, it should be copied when derive stats

Reply:
I think statistics describe the statistics of the original table, you can store the original statistics and convert them to numeric values when they are used elsewhere. I'm not quite sure how other databases do it? Are their stats like that too?

Reply:
I will mark all the field of ColumnStat and TabletStat as `final` in another PR, since this PR is  a little huge now. I think this discussion could be ended. If you have any other question, you could open a issue or propose it on the weekly discussion.

Reply:
@morrySnow

Reply:
if the name is a problem, let's use `statistics` instead of 'stats` or 'stat`.